bftengine/src/bcstatetransfer/Messages.hpp

```diff
@@ -256,7 +239,8 @@ struct ItemDataMsg : public BCStateTranBaseMsg {
                             // digests from the total of dataSize (rvbDigestsSize < dataSize)
   char data[1];             // MSB[raw block of size dataSize-rvbDigestsSize|RVB DIGESTS of size rvbDigestsSize]LSB
 
-  uint32_t size() const { return sizeof(ItemDataMsg) - 1 + dataSize; }
+  uint32_t size() const { return MsgAllocator<ItemDataMsg>::calcBufferSize(dataSize); }
+  void setDataSize(uint32_t size) { dataSize = size; }
```

Comment:
same as my comment before. I would use a friend class (yes, it's better) and not expose a new public api which allow anyone from outside to change data size.